runtimerevolution/web3-roulette
# [WEB3-25] Participant management endpoint

# Description

When it is not possible to verify giveaway requirements, the participant is added with a pending state. This PR adds an endpoint to update participant state in a given giveaway. Using this endpoint the user can switch to a confirmed, or rejected, participation state.

Ticket: https://runtime-revolution.atlassian.net/browse/WEB3-25

## Type of change

- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Jest tests were added to ensure the endpoint functionality:
- ensure that the participant state can successfully be updated
- ensure that the participant is not added to the smart contract when it is rejected
- ensure bad request responses on invalid participants / invalid states

## Checklist:
